src/main/daml/Daml/Finance/Instrument/Generic/Util.daml

```diff
@@ -62,6 +62,25 @@ createFixRatePaymentClaims schedule periodicSchedule useAdjustedDatesForDcf coup
   prepareAndTagClaims couponClaims "Fix rate payment"
 -- FIXED_RATE_BOND_COUPON_CLAIMS_END
 
+-- | Calculate a fix rate amount (if a credit event has not yet happened) for each payment date and create claims.
+createConditionalCreditFixRatePaymentClaims : Applicative f => [SchedulePeriod] -> PeriodicSchedule -> Bool -> Decimal -> Decimal -> DayCountConventionEnum -> Deliverable -> Observable -> f [TaggedClaim]
+createConditionalCreditFixRatePaymentClaims schedule periodicSchedule useAdjustedDatesForDcf couponRate weightMultiplier dayCountConvention cashInstrumentCid defaultProbabilityReferenceId = do
+  let
+    creditEvent = Lte (Const 1.0, Observe defaultProbabilityReferenceId)
+    couponDatesAdjusted = map (.adjustedEndDate) schedule
+    couponAmounts = map (\p -> couponRate * (calcPeriodDcf dayCountConvention p useAdjustedDatesForDcf periodicSchedule.terminationDate periodicSchedule.frequency)) schedule
+    couponClaims = zipWith (\d a -> until creditEvent $ when (TimeGte $ d) $ scale (Const weightMultiplier * Const a) $ one cashInstrumentCid) couponDatesAdjusted couponAmounts
+  prepareAndTagClaims couponClaims "Fix rate payment (unless credit event has occurred)"
+
+-- | Calculate a (1-recoveryRate) payment if a credit event just happened and create claims.
+createCreditEventPaymentClaims : Applicative f => Decimal -> Deliverable -> Observable -> Observable -> PeriodicSchedule -> f [TaggedClaim]
+createCreditEventPaymentClaims weightMultiplier cashInstrumentCid defaultProbabilityReferenceId recoveryRateReferenceId periodicSchedule = do
+  let
+    creditEvent = Lte (Const 1.0, Observe defaultProbabilityReferenceId)
+    payoff = scale (Const weightMultiplier * (Const 1.0 - Observe recoveryRateReferenceId)) $ one cashInstrumentCid
+    creditEventClaims = [until (TimeGte $ periodicSchedule.terminationDate) $ until creditEvent $ when (TimeGte $ periodicSchedule.effectiveDate) $ when creditEvent $ payoff]
```

Comment:
I think with the correct semantics the sub-tree under `until creditEvent` would expire to `zero` when the event happens. So maybe it should rather be `when (>= startDate) $ until (>= endDate) $ when creditEvent payoff`. But keen to hear @mlimberto 's thoughts here.

Reply:
I changed it to that order for the time being (including `until creditEvent` as well so that the instrument can only pay the credit protection amount once).

Reply:
Once we have implemented the `expire` logic it will be easier to see the effects of the two approaches. For now, the tests of the two CDSs both pass.

Reply:
> I think with the correct semantics the sub-tree under `until creditEvent` would expire to `zero` when the event happens. So maybe it should rather be `when (>= startDate) $ until (>= endDate) $ when creditEvent payoff`. But keen to hear @mlimberto 's thoughts here.

I second Georg's suggestion

Reply:
you don't actually need the `until creditEvent` on this leg, as the `when creditEvent` will expire when the sub-tree underneath it is acquired (ie. when you lifecycle it first after the condition is true). so there's no risk of paying twice here)

Reply:
I see, the lifecycling takes care of expiring the `when`. Thanks for the explanation!